tools/cloud-build/daily-tests/ansible_playbooks/tasks/create_deployment_directory.yml

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Assert variables are defined
+  ansible.builtin.assert:
+    that:
+    - blueprint_yaml is defined
+    - project is defined
+    - workspace is defined
+    - deployment_name is defined
+    - test_name is defined
+
+- name: Create cli flag for extra deployment variables
+  ansible.builtin.set_fact:
+    deployment_vars_str: "--vars {{ cli_deployment_vars.items() | map('join', '=') | join(',') }}"
+  when: cli_deployment_vars is defined
+
+- name: Create Blueprint
+  ansible.builtin.command: |
+      ./ghpc create -l ERROR "{{ blueprint_yaml }}" \
+      --backend-config bucket=daily-tests-tf-state \
```

Comment:
This is exposed via the `GHPC_DEV_BUCKET` variable in the existing script and while that appears to go unused, I think it would be better to avoid duplication and facilitate setting this via secret manager or similar.

I suggest something like

```yaml
- name: Set bucket
  ansible.builtin.set_fact:
    state_bucket: "{{ state_bucket | default(daily-tests-tf-state) }}"
```

This enables `state_bucket` to be set all the way back at `daily-tests/builds/*`.

You might benefit from single quotes around the whole thing and double quotes around the bucket default value.

Reply:
Discussed offline. My opinion is that we have not used this option in all the time it has been available. It is very easy to add this functionality if we transition this value to secret manager.

I do agree with your point that the value should be defined in a single place. I have set is as a fact and used the fact in subsequent tasks.